File: src/main/java/com/mallang/member/query/dao/support/MemberQuerySupport.java
package com.mallang.member.query.dao.support;

import com.mallang.member.domain.Member;
import java.util.NoSuchElementException;
import org.springframework.data.jpa.repository.JpaRepository;

public interface MemberQuerySupport extends JpaRepository<Member, Long> {

    default Member getById(Long id) {
        return findById(id).orElseThrow(() -> new NoSuchElementException(
                "id가 %d인 회원을 찾을 수 없습니다.".formatted(id)
        ));
    }
}

File: src/main/kotlin/com/mallang/auth/query/dao/support/MemberQuerySupport.kt
package com.mallang.auth.query.dao.support

import com.mallang.auth.domain.Member
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.findByIdOrNull


interface MemberQuerySupport : JpaRepository<Member, Long?> {
    fun getById(id: Long) = findByIdOrNull(id)
            ?: throw NoSuchElementException("id가 ${id}인 회원을 찾을 수 없습니다.")
}
